core/network/protocol_base.hpp

```diff
@@ -27,7 +27,9 @@ namespace kagome::network {
     ProtocolBase &operator=(ProtocolBase &&) noexcept = delete;
     ProtocolBase &operator=(ProtocolBase const &) = delete;
 
-    virtual const Protocol &protocol() const = 0;
+    virtual const std::string &protocolName() const {
+      return protocol_name_;
```

Comment:
I suggest to remove protocol_name_ variable and just make this function as virtual pure one and implement in inherited classes in order: 
1. to be consistent with other functions in this file
2. not to rely on developers who would need to initialize `protocol_name_` variable in constructor's body of inherited classes (they might forget to do so when introduce new constructor for example)